feature-libs/asm/components/asm-customer-360/sections/asm-customer-activity/asm-customer-activity.component.ts

```diff
@@ -1,128 +1,253 @@
+      this.orderHistoryFacade.getOrderHistoryList(this.ORDER_LIMIT),
+      this.activeCartFacade.getActive(),
+      this.savedCartFacade.getList(),
+    ]).pipe(
+      switchMap(([config, orderHistory, activeCart, savedCarts]) => {
+        this.pageSize = config.pageSize || this.PAGE_SIZE;
+        entries = [];
+        // notes: active cart does not have date
+        if (activeCart) {
+          entries.push({
+            typeId: 'activeCart',
+            id: activeCart.code,
+            created: undefined,
+            category: '',
+          });
+          this.saveLocalization(
+            activeCart.code,
+            'type',
+            'asm.customer360.activity.cart'
+          );
+          this.saveLocalization(
+            activeCart.code,
+            'description',
+            'asm.customer360.activity.numberOfCartItems',
+            {
+              count: activeCart.totalItems ?? 0,
+            }
+          );
+        }
+
+        if (savedCarts) {
+          savedCarts.forEach((cart) => {
+            entries.push({
+              typeId: 'savedCart',
+              id: cart.code,
+              created: cart?.saveTime
+                ? Date.parse(String(cart?.saveTime))
+                : undefined,
+              category: '',
+            });
+            this.saveLocalization(
+              cart.code,
+              'type',
+              'asm.customer360.activity.savedCart'
+            );
+            this.saveLocalization(
+              cart.code,
+              'description',
+              'asm.customer360.activity.numberOfCartItems',
+              {
+                count: cart.totalItems ?? 0,
+              }
+            );
+          });
+        }
+        // Notes: order history order doesn't have totalItems
+        if (orderHistory?.orders?.length) {
+          orderHistory.orders.forEach((order) => {
+            entries.push({
+              typeId: 'orderHistory',
+              id: order.code,
+              created: order?.placed
+                ? Date.parse(String(order?.placed))
+                : undefined,
+              category: order.statusDisplay,
+            });
+            this.saveLocalization(
+              order.code,
+              'type',
+              'asm.customer360.activity.order'
+            );
+          });
+        }
+        return this.getLocalizations().pipe(
+          map((valueLocalizations) => {
+            valueLocalizations.forEach((valueLocalization) => {
+              const entry = entries.find(
+                (item) => item.id === valueLocalization.id
+              );
+              if (entry && valueLocalization.propertyName) {
+                entry[valueLocalization.propertyName] = valueLocalization.value;
+              }
+            });
+            return entries;
+          })
+        );
+      })
+    );
+    this.savedCartFacade.loadSavedCarts();
+  }
+
+  itemSelected(entry: GeneralEntry): void {
```

Comment:
`entry` can be `undefined`? Odd.
```suggestion
  itemSelected(entry: GeneralEntry | undefined): void {
```